components/list/list-item-checkbox-mixin.js

```diff
@@ -56,6 +56,19 @@ export const ListItemCheckboxMixin = superclass => class extends SkeletonMixin(L
 		this._checkboxId = getUniqueId();
 	}
 
+	get selectionInfo() {
+		return this._selectionInfo;
+	}
+
+	set selectionInfo(val) {
+		const oldVal = this._selectionInfo;
+		if (oldVal !== val) {
+			this._selectionInfo = val;
+			this.selected = (this._selectionInfo.state === SelectionInfo.states.all);
```

Comment:
Not delaying the event caused some trouble with the selection behaviour of list-items where a parent list-item (of a nested list) would become selected and then unselected because the `SelectionMixin` updates `selectionInfo` of observers (i.e. parent list-items in this case) causing a rerender while the selected state is false, thereby undoing the selection state of the `d2l-selection-input` it renders. 

This change ensures that when the `SelectionMixin` updates the `selectionInfo` property of the list-item, the `selected` property is also updated so that when the list-item re-renders, it does not incorrectly set the `d2l-selection-input` `selected` state back to false.